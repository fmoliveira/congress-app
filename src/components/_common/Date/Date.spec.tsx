import React from "react"
import { cleanup, render } from "react-testing-library"

import { Date } from "./index"

const setup = (props?: any) => render(<Date {...props} />)

const SAMPLE_DATE = "Mon, 4 Feb 2019 15:16:17 EST"

describe("Date", () => {
  afterEach(() => {
    cleanup()
  })

  it("Should not render without a value", () => {
    const { container } = setup()
    expect(container.firstChild).toBeNull()
    expect(container).toMatchSnapshot()
  })

  it("Should render with a value", () => {
    const props = { value: SAMPLE_DATE }
    const { container, getByText } = setup(props)
    expect(container.firstChild).toBeTruthy()
    expect(getByText("2019-02-04")).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it("Should render with a custom format", () => {
    const props = { value: SAMPLE_DATE, displayFormat: "MM/DD/YYYY" }
    const { container, getByText } = setup(props)
    expect(container.firstChild).toBeTruthy()
    expect(getByText("02/04/2019")).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it("Should render with a time format", () => {
    const props = { value: SAMPLE_DATE, displayFormat: "MM/DD/YYYY h:mm A" }
    const { container, getByText } = setup(props)
    expect(container.firstChild).toBeTruthy()
    expect(getByText("02/04/2019 8:16 PM")).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
})
