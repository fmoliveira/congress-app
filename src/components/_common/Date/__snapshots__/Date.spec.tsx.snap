// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Should not render without a value 1`] = `<div />`;

exports[`Date Should render with a custom format 1`] = `
<div>
  <time
    datetime="Mon, 4 Feb 2019 15:16:17 EST"
  >
    02/04/2019
  </time>
</div>
`;

exports[`Date Should render with a time format 1`] = `
<div>
  <time
    datetime="Mon, 4 Feb 2019 15:16:17 EST"
  >
    02/04/2019 8:16 PM
  </time>
</div>
`;

exports[`Date Should render with a value 1`] = `
<div>
  <time
    datetime="Mon, 4 Feb 2019 15:16:17 EST"
  >
    2019-02-04
  </time>
</div>
`;
